cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -1023,11 +1019,11 @@ impl From<StackAMode> for SyntheticAmode {
         // We enforce a 128 MB stack-frame size limit above, so these
         // `expect()`s should never fail.
         match amode {
-            StackAMode::FPOffset(off, _ty) => {
+            StackAMode::ArgOffset(off, _ty) => {
                 let off = i32::try_from(off)
                     .expect("Offset in FPOffset is greater than 2GB; should hit impl limit first");
```

Comment:
```suggestion
                    .expect("Offset in ArgOffset is greater than 2GB; should hit implementation limit first");
```

Reply:
The `+ 16` to compute the final frame pointer offset can now overflow too, right?

Reply:
Good catches! With regard to overflow, I'm moving this `+16` inside the `i32::try_from` so the add happens at `i64` instead. That makes its overflow behavior the same as current, which already had unchecked `i64` addition. Similarly, the aarch64 and riscv64 targets are still doing the addition at `i64` after this patch, and s390x doesn't add anything so can't overflow.